include/copc-lib/io/reader.hpp

```diff
@@ -53,5 +56,24 @@ class Reader : public BaseIO
 
     std::vector<Entry> ReadPage(std::shared_ptr<Internal::PageInternal> page) override;
 };
+
+class FileReader : public Reader
+{
+  public:
+    FileReader(const std::string &file_path)
+    {
+        in_stream_ = new std::fstream;
```

Comment:
you can do what you did before, so we can avoid the cast:

fstream* stream = new std::fstream();
stream->open();
in_stream_ = stream;